src/subdomains/supporting/bank-tx/bank-tx/services/sepa-parser.service.ts

```diff
@@ -120,31 +120,57 @@ export class SepaParser implements OnModuleInit {
     chargeAmount: number;
     chargeAmountChf: number;
     chargeCurrency: string;
+    senderChargeAmount: number;
+    senderChargeAmountChf: number;
+    senderChargeCurrency: string;
   }> {
     let chargeAmount = 0;
     let chargeAmountChf = 0;
     const chargeCurrencies = new Set<string>();
 
-    if (!charges) return { chargeAmount, chargeAmountChf, chargeCurrency: Config.defaults.currency };
+    let senderChargeAmount = 0;
+    let senderChargeAmountChf = 0;
+    const senderChargeCurrencies = new Set<string>();
+
+    if (!charges)
+      return {
+        chargeAmount,
+        chargeAmountChf,
+        chargeCurrency: Config.defaults.currency,
+        senderChargeAmount,
+        senderChargeAmountChf,
+        senderChargeCurrency: Config.defaults.currency,
+      };
 
     charges = (Array.isArray(charges) ? charges : [charges]).filter((c) => +c.Amt['#text'] !== 0);
 
+    const referenceAsset = await this.fiatService.getFiatByName(charges[0].Amt['@_Ccy']);
+
```

Comment:
Don't we want the charge amount in the sender asset? It's not really defined, what the currency of the first charge is, is it?

Reply:
Yeah good point, will change that 👍